### PR TITLE
(fix): clarify public backed sparse docstring/api

### PR DESCRIPTION
**Pull request by ilan-gold**
For https://github.com/scverse/anndata/pull/1598#pullrequestreview-2260162414

Perhaps should go in separately to `main` as it is a cleaner git blame IMO
- [ ] Closes #
- [ ] Tests added
- [ ] Release note added (or unnecessary)

### CODE REVIEW

**Comment by flying-sheep** [comment]
So you want us to commit to the whole API?

If not necessarily: There are ways to make users see only part of it, I can help with that.

**Comment by ilan-gold** [comment]
> If not necessarily: There are ways to make users see only part of it, I can help with that.

Sorry went to lunch before getting to that.  I made private what I thought should be, and left what I thought should be public as such.  Feel free to comment otherwise

**Review by flying-sheep — APPROVED** [review]
Awesome! I don’t think we want people to manually initialize it, but I think not much harm is done if they try do do that.

File: src/anndata/_core/sparse_dataset.py [hunk 1]
```diff
@@ -569,21 +572,29 @@ def _to_backed(self) -> BackedSparseMatrix:
...
 
-    def to_memory(self) -> ss.csr_matrix | ss.csc_matrix:
+    def to_memory(self) -> spmatrix | SpArray:
+        """Returns an in-memory representation of the sparse matrix.
+
+        Returns
+        -------
+            The in-memory representation of the sparse matrix.
+        """
         format_class = get_memory_class(self.format)
```
Comment (flying-sheep) [inline src/anndata/_core/sparse_dataset.py:585]: This never returns `csr_array`/`csc_array`, should it?

Comment (ilan-gold) [reply src/anndata/_core/sparse_dataset.py:585]: At the moment, it shouldn't.  But we need at add a setting in for that, thanks for the reminder.   Will PR now

Comment (flying-sheep) [reply src/anndata/_core/sparse_dataset.py:585]: OK, then that doesn’t stop this PR

**Review by flying-sheep — APPROVED** [review]

File: src/anndata/_core/sparse_dataset.py [hunk 2]
```diff
@@ -569,21 +572,29 @@ def _to_backed(self) -> BackedSparseMatrix:
...
+        mtx.indptr = self._indptr
         return mtx
 
-    def to_memory(self) -> ss.csr_matrix | ss.csc_matrix:
+    def to_memory(self) -> spmatrix | SpArray:
+        """Returns an in-memory representation of the sparse matrix.
+
+        Returns
+        -------
+            The in-memory representation of the sparse matrix.
```
Comment (flying-sheep) [inline src/anndata/_core/sparse_dataset.py:583]:
```suggestion
        The in-memory representation of the sparse matrix.
```

**Pull request merged**